rmw_zenoh_cpp/src/detail/liveliness_utils.cpp

```diff
@@ -167,48 +166,113 @@ std::vector<std::string> split_keyexpr(
   result.push_back(keyexpr.substr(start));
   return result;
 }
+
+///=============================================================================
+// Helper function to convert string to size_t.
+// The function is templated to enable conversion to size_t or std::size_t.
+template<typename T>
+std::optional<T> str_to_size_t(const std::string & str, const T default_value)
+{
+  if (str.empty()) {
+    return default_value;
+  }
+  errno = 0;
+  char * endptr;
+  // TODO(Yadunund): strtoul and strtol both return long int and not size_t so we
+  // should consider fixing this implementation if this function is moved to
+  // a header file and will be used by other parts of the codebase.
+  size_t num = std::is_signed<decltype(default_value)>::value ?
+    strtol(str.c_str(), &endptr, 10) :
+    strtoul(str.c_str(), &endptr, 10);
```

Comment:
So this is a bit trickier, because `strtol` returns a `long` (signed), and `strtoul` returns an `unsigned long` (unsigned).  Thus, it isn't totally correct to assign them both to a `size_t`, particularly the signed long.

I'm going to suggest that for now, we just leave this as it was and add the TODO comment:

```suggestion
  // TODO(Yadunund): strtoul returns an unsigned long, not size_t.
  // Depending on the architecture and platform, these may not be the same size.
  // Further, if the incoming str is a signed integer, storing it in a size_t is incorrect.
  // We should fix this piece of code to deal with both of those situations.
  size_t num = strtoul(str.c_str(), &endptr, 10);
```